upstream/commands.go

```diff
@@ -168,9 +168,10 @@ func InsertUpstreamMsg(ctx context.Context, req *PushData) error {
 		}
 	}
 
-	// config items are inserted one by one, instead of in a batch, because of the foreign key constraint with itself.
-	if err := saveIndividuallyWithRetries(ctx, req.ConfigItems, saveRetries); err != nil {
-		return fmt.Errorf("error upserting components: %w", err)
+	if len(req.ConfigItems) > 0 {
+		if err := db.Clauses(clause.OnConflict{UpdateAll: true}).CreateInBatches(req.ConfigItems, batchSize).Error; err != nil {
```

Comment:
I think this needs to be wrapped ina transaction with a `SET CONSTRAINTS ALL DEFERRED`

Reply:
```
Foreign-key constraints:
    "config_items_agent_id_fkey" FOREIGN KEY (agent_id) REFERENCES agents(id)
    "config_items_parent_id_fkey" FOREIGN KEY (parent_id) REFERENCES config_items(id) DEFERRABLE INITIALLY DEFERRED
    "config_items_scraper_id_fkey" FOREIGN KEY (scraper_id) REFERENCES config_scrapers(id)
```

I think that's not required since it's differed initially by default?